modules/text/src/ocr_holistic.cpp

```diff
@@ -0,0 +1,1064 @@
+        transcription=this->labels_[classNum];
+    }
+
+    void recogniseImageBatch(InputArrayOfArrays inputImageList,CV_OUT std::vector<String>& transcriptionVec,CV_OUT std::vector<double>& confidenceVec)
+    {
+        Mat netOutput;
+        this->classifier_->classifyBatch(inputImageList,netOutput);
+
+        for(int k=0;k<netOutput.rows;k++)
+        {
+            int classNum;
+            double confidence;
+            NetOutput::getClassification((float*)(netOutput.row(k).data),this->classifier_->getOutputSize(),classNum,confidence);
+            transcriptionVec.push_back(this->labels_[classNum]);
+            confidenceVec.push_back(confidence);
+        }
+    }
+
+
+    void run(Mat& image, std::string& output_text, std::vector<Rect>* component_rects=NULL,
+             std::vector<std::string>* component_texts=NULL, std::vector<float>* component_confidences=NULL,
+             int component_level=0)
+    {
+        CV_Assert(component_level==OCR_LEVEL_WORD);//Componnents not applicable for word spotting
+        double confidence;
+        String transcription;
+        recogniseImage(image,transcription,confidence);
+        output_text=transcription.c_str();
+        if(component_rects!=NULL)
+        {
+            component_rects->resize(1);
+            (*component_rects)[0]=Rect(0,0,image.size().width,image.size().height);
+        }
+        if(component_texts!=NULL)
+        {
+            component_texts->resize(1);
+            (*component_texts)[0]=transcription.c_str();
+        }
+        if(component_confidences!=NULL)
+        {
+            component_confidences->resize(1);
+            (*component_confidences)[0]=float(confidence);
+        }
+    }
+
+    void run(Mat& image, Mat& mask, std::string& output_text, std::vector<Rect>* component_rects=NULL,
+             std::vector<std::string>* component_texts=NULL, std::vector<float>* component_confidences=NULL,
+             int component_level=0)
+    {
+        CV_Assert(mask.cols==image.cols && mask.rows== image.rows);//Mask is ignored because the CNN operates on a full image
+        this->run(image,output_text,component_rects,component_texts,component_confidences,component_level);
+    }
+
+    std::vector<String>& getVocabulary()
+    {
+        return this->labels_;
+    }
+
+    Ptr<TextImageClassifier> getClassifier()
+    {
+        return this->classifier_;
+    }
+};
+
+Ptr<OCRHolisticWordRecognizer> OCRHolisticWordRecognizer::create(Ptr<TextImageClassifier> classifierPtr,String vocabularyFilename )
+{
+    return Ptr<OCRHolisticWordRecognizer>(new OCRHolisticWordRecognizerImpl(classifierPtr,vocabularyFilename));
+}
+
+Ptr<OCRHolisticWordRecognizer> OCRHolisticWordRecognizer::create(String modelArchFilename, String modelWeightsFilename, String vocabularyFilename)
+{
+    Ptr<ImagePreprocessor> preprocessor=ImagePreprocessor::createImageStandarizer(113);
```

Comment:
How do you obtain 113 as a parameter? What does it mean? Can it be a named constant?

Reply:
This is part of code written as part of GSOC'16 by Anguelos, @lluisgomez  or @anguelos can throw some light.